### PR TITLE
[13] [UI] Skeleton loading screen

### PR DESCRIPTION
**Pull request by AVI5HEK**
https://github.com/nimblehq/ic-flutter-avishek/issues/13

## What happened 👀

To let the user know the application is fetching data and keep them occupied, we need to show the loading animation on the Home screen when the app is fetching surveys.

## Insight 📝

N/A.

### Android

https://github.com/nimblehq/ic-flutter-avishek/assets/8093908/7050933b-d4db-4377-a265-08b91eb999c4

### iOS

https://github.com/nimblehq/ic-flutter-avishek/assets/8093908/4e4523a6-2458-4099-8e14-4add2f1290df

### CODE REVIEW

File: lib/ui/home/home_screen.dart [hunk 1]
```diff
@@ -56,10 +57,7 @@ class HomeScreenState extends ConsumerState<HomeScreen> {
     final uiModels = ref.watch(_surveysStreamProvider).value ?? [];
     return ref.watch(homeViewModelProvider).when(
           init: () => const Center(
-            // TODO: replace CircularProgressIndicator with the shimmer effect.
-            child: CircularProgressIndicator(
-              valueColor: AlwaysStoppedAnimation(Colors.white),
-            ),
+            child: SkeletonLoadingScreen(),
```
Comment (Wadeewee) [inline lib/ui/home/home_screen.dart:60]: @AVI5HEK I think we can remove `const Center` and use `const SkeletonLoadingScreen()` instead

**Review by Wadeewee — COMMENTED** [review]
For the rest LGTM 🚀

File: lib/ui/home/home_screen.dart (hunk 1 above)
Comment (AVI5HEK) [reply lib/ui/home/home_screen.dart]: @Wadeewee done in https://github.com/nimblehq/ic-flutter-avishek/pull/46/commits/750980c926f3cbbe217abcb70c953733a3e6dfb0

**Review by Wadeewee — APPROVED** [review]

File: lib/ui/widget/skeleton_loading_screen.dart [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+              _buildSkeleton(width: screenWidth / 1.5),
+              const SizedBox(height: 54),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Row buildHeaderSkeleton(double screenWidth) {
```
[the remark this answers is not in the document]
Comment (AVI5HEK) [reply lib/ui/widget/skeleton_loading_screen.dart]: Resolved here https://github.com/nimblehq/ic-flutter-avishek/pull/46/commits/1fcb01689a94a819f763061630346074f698b16f

**Review by markgravity — APPROVED** [review]

**Pull request merged**